api/fixtures_test.go

```diff
@@ -0,0 +1,490 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+
+	"github.com/algorand/go-algorand/ledger"
+
+	"github.com/algorand/indexer/api/generated/v2"
+	"github.com/algorand/indexer/idb/postgres"
+	"github.com/algorand/indexer/processor"
+	"github.com/algorand/indexer/util/test"
+)
+
+/* **************************************
+This Fixtures Test simulates calling indexer GET endpoints and in so doing validates:
+* Hand crafted URL's are:
+	* routed to the correct endpoint
+	* handled as expected
+* HTTP Responses are:
+	* well formatted and can be parsed to produce results of the expected type
+	* have the state that is expected for a "real" blockchain
+
+This is achieved as follows:
+A. Set up the state of the blockchain
+B. Iterate through a hard-coded `xyzSeedFixture` to generate an `xyzLiveFixture` as follows. For each seed object in `xyzSeedFixture`:
+	1. Query indexer via HTTP for the state of the blockchain using the `seed`'s `Request.Path` + `Request.Params`
+	2. Query the test-internal `proverRoutes` struct for a route-compatible `prover`
+	3. Employ the `prover` to parse the indexer's response into a route appropriate `witness` object
+C. Save `xyzLiveFixture` into a non-git-committed fixture `./test_resources/_FIXTURE_NAME.json` (NOTICE THE `_` PREFIX)
+D. Assert that the non-git-commited fixture equals the git-committed version `./test_resources/FIXTURE_NAME.json`
+
+It is the responsibility of the test writer to check that the generated fixture represent the expected results.
+************************************** */
```

Comment:
This seems really complicated. I've spent a bit of time trying to understand how the components go together and am still having trouble.

Could you help me understand the benefits to this system vs the others that we have?

A few particular points that threw me off so far (maybe this is just me):
* `xyz` as a wildcard, I couldn't find `xyzSeedFixture` for a while, eventually I notice `*SeedFixture` functions.
* Using the word `Proof` and a function `getProof` in such close proximity to `StateProofs` and it's own `GetProof` operation.
* generally, there are a lot of moving pieces and I'm having trouble distinguishing between the fixtures and the tests.

Reply:
Am resolving per our live discussion. I have committed to better documentation for how to create a fixtures test and put that commitment in the form of a TODO in the PR description.